### DISCUSSION TITLE
how print message in two ReplyKeyboardMarkup

### DISCUSSION THREAD

**Discussion by hossain93** [aiogram/aiogram#854] [Q&A]
how I can print in any handlers when  I enter my words

```python
@dp.message_handler(lambda message: message.text == "AC")
async def registration(message: types.Message):
   keyboard = types.ReplyKeyboardMarkup(resize_keyboard=True)
   buttons = ["A","B"]
   keyboard.add(*buttons)
   await message.answer("click", reply_markup=keyboard)


@dp.message_handler(lambda message: message.text == "A")
async def registration(message: types.Message):
       keyboard = types.ReplyKeyboardMarkup(resize_keyboard=True)
       buttons = ["B ","A "]
       keyboard.add(*buttons)
       await message.answer("enter your text", reply_markup=keyboard)
       @dp.message_handler(regexp='[\w|\W]')
       async def printtext(message: types.update):
          return print("***")    


@dp.message_handler(lambda message: message.text == "B")
async def registration(message: types.Message):
   keyboard = types.ReplyKeyboardMarkup(resize_keyboard=True)
   buttons = ["done"]
   keyboard.add(*buttons)
   await message.answer("enter your text", reply_markup=keyboard)
   @dp.message_handler(regexp='[\w|\W]' and )
   async def savetext(message: types.update):
         return print("+++")
```

when I run this, just print the first function that I click on that. for example when I click on A and then enter a text, my code print *** 
but when I click on B and enter a text, again print *** while I want print +++

**Comment by evgfilim1** [comment]
Have a look at FSM: https://github.com/aiogram/aiogram/blob/dev-2.x/examples/finite_state_machine_example.py

**Reply by hossain93** [reply]
> Have a look at FSM: https://github.com/aiogram/aiogram/blob/dev-2.x/examples/finite_state_machine_example.py

I don't want to use this method

**Reply by evgfilim1** [reply]
> > Have a look at FSM: https://github.com/aiogram/aiogram/blob/dev-2.x/examples/finite_state_machine_example.py
> 
> I don't want to use this method

Then I can't help you. Any kind of dialogs are built only on FSM.

**Reply by JrooTJunior** [reply]
> I don't want to use this method

Why? Whats your motivation?

**Comment by JrooTJunior** [comment]
handlers cannot be registered inside handlers, as in your example.

All registered handlers in Dispatcher is available globally to all bot users, and events can only be routed only if this handler is registered. 

In youre case inner handlers (registered inside handlers) will be available only aftrer outer handler is called and and each time the handler is called, will register again and again until bot stops and it meakes and causes memory leak with an uncontrolled flow of bot events.

So.. You should never register handlers from inside handlers.

There is only only one correct answer for your question - use finite state machine (FSM). This can be FSM integrated into agioram or any other FSM solution.
